designer/server/src/lib/forms.js

```diff
@@ -55,6 +62,34 @@ export async function remove(id) {
 }
 
 /**
+ * Get draft form definition
+ * @param {string} id
+ */
+export async function getDraftFormDefinition(id) {
+  const endpoint = getDraftFormDefinitionEndpoint(id)
+  const getJsonByType = /** @type {typeof getJson<FormDefinition>} */ (getJson)
+  const { body } = await getJsonByType(endpoint)
```

Comment:
Looks good for now whilst we get things working

Just a note to review parameter misuse in another PR, e.g.

```js
// Hostname swap
id = '@example.org'

// Path traversal
id = '../secrets'

// TODO: Swap with new URL() constructor
const endpoint = `${formsEndpoint}/${id}/definition/draft`;
```

One for another PR, but adding a trailing slash to `formsEndpoint` and using [`new URL()` with base param](https://nodejs.org/docs/latest-v20.x/api/url.html#new-urlinput-base) instead of strings will help us do this

```suggestion
  const getJsonByType = /** @type {typeof getJson<FormDefinition>} */ (getJson)
  const { body } = await getJsonByType(
    new URL(`${id}/definition/draft`, formsEndpoint)
  )
```